service/dap/server_test.go

```diff
@@ -2060,7 +2060,13 @@ func TestNextAndStep(t *testing.T) {
 					client.ExpectStepInResponse(t)
 					expectStop("main.inlineThis", 5)
 
-					client.NextRequest(-10000 /*this is ignored*/)
+					client.NextRequest(-1000)
+					client.ExpectNextResponse(t)
+					if se := client.ExpectStoppedEvent(t); se.Body.Reason != "error" || se.Body.Text != "unknown goroutine -1000" {
+						t.Errorf("got %#v, want Reaspon=\"error\", Text=\"unknown goroutine -1000\"", se)
+					}
+
+					client.NextRequest(1)
```

Comment:
I think it is better not to add extra next and just change last expectStop to the previous line 5.

Reply:
I would also add the follow to verify that we reported a stop on the same thread at the same function location
```handleStop(t, client, 1, "main.inlineThis", 6)```

Reply:
This would fail without `client.NextRequest(1)`.

Reply:
How does it fail? It shouldn't fail as far as I can tell.
If you want, you can add a TODO for me to sort this out later.

Reply:
It should be handleStop(t, client, 1, "main.inlineThis", 5) - the line from the previous next stop, since this one did not succeed.

Reply:
The stop event returned by NextRequest(-1000) is already consumed by the ExpectStoppedEvent that checks for the error.

Reply:
I think you are confusing two different helpers - expectStop and handleStop. expectStop = ExpectStoppedEvent+checks+handleStop. I am proposing that you add handleStop, which won't try to consume any additional stopped events.

Reply:
Got it.